README.md

```diff
@@ -295,14 +295,19 @@ The `task_state` field takes one of the following values
 (different queue types support different
 sets of `task_state` values, so this is a superset):
 
-* 'r' - the task is ready for execution (the first consumer executing
+* 'r' - the task is **ready** for execution (the first consumer executing
 a `take` request will get it)
-* 't' - the task has been taken by a consumer
-* '-' - the task has been executed (a task is removed from the queue
-after it
-   has been executed, so this value will rarely be seen)
-* '!' - the task is buried (disabled temporarily until further changes)
-* '~' - the task is delayed for some time
+* 't' - the task has been **taken** by a consumer
+* '-' - the task has been **executed (done)** (a task is removed from the queue
+after it has been executed, so this value will rarely be seen)
+* '!' - the task is **buried** (disabled temporarily until further changes)
+* '~' - the task is **delayed** for some time.
+
+For better understanding of the task life cycle, below is the task state
+diagram with indication of all transitions between the tasks states mentioned
+above and triggers of these transitions.
+
+![Task state diagram](images/task_state_diag.png)
```

Comment:
Please, save source text of the diagram within the repository and give pointers how to render the image.

Possible directory structure:

```
+ doc/
  + images/
    - statechart.png
    + src/
      - statechart.puml
      - README.md
```

I propose to describe steps to generate `doc/images/statechart.png` in `doc/images/src/README.md` or right within `doc/images/src/statechart.puml` in a comment if possible. Don't forget to include URL to the webpage if the process involves usage of an online tool.

Reply:
Alternative to saving .png to the repository is using of an external rendering service: https://blog.anoff.io/2018-07-31-diagrams-with-plantuml/#github-integration

Reply:
The alternative way you mentioned works. I think we should use it because it's more convenient: you just need to keep the source plantuml code in `/doc/scr/` and include a link to the PlantUML rendering service in readme.md via the standard markdown syntax `![]()`.
To update the diagram, you need to change just the source plantuml code.
If you think we still need a kind of readme for this case, I will add it.

Reply:
Nope, if we'll use this way, everything should just work and doc/src/README.md is not necessary.

Reply:
@Totktonada I've added a GH Action to (re)generate and save diagrams. Please have a look.

Reply:
This way we store artifacts on our side and don't depend on reliability of an external service. That's good.

There are several downsides:

* Unclear workflow: an extra commit will be added into the PR, but then one more extra commit will be added into `master`?
* A generated file in the repository.

The first one disquiets me, to be honest. I don't want touching of my branch (and, moreover, `master`) by a robot. Version control systems are for humans.

Since @LeonidVas maintaining the repository, this will be his decision.

Reply:
OK. Decided to use the option of saving the diagram file (png or svg) in the repository [#145 (comment)](https://github.com/tarantool/queue/pull/145#discussion_r553206221). The option of using some external rendering service seems not very reliable.

Reply:
Put the diagram here: ./doc/image/statediagram.svg
Wrote a short instruction on rendering diagram in SVG using https://www.planttext.com/. Put the instruction in statediagram.pulm as comments as Alexander asked.